src/index.js

```diff
@@ -19,664 +20,520 @@ const debug = (function () {
-    this.stayConnected = false
-    this.loginSw = null
-    this.loginPeer = null
-
-    // reset replication variables
-    this.replicating = false
-    this.userAppRepSW = null
-    this.userAppRepHub = null
-
-    // reset logged in user variable
-    this.userAppDbId = null
-    this.userAppDb = null
-    this.userAppDEK = null
-    this.importedUserAppDEK = null
-    this.userAppNonce = null
-    this.username = null
-    this.profileImage = null
   }
 
   async init () {
-    await this.initPromise
-  }
-
-  async _init () {
-    debug('[masq._init]')
-    // try to login from session info
-    await this._loadSessionInfo()
-
-    // get login channel and key created at login link generation
-    // sets this.loginKey and this.loginChannel
-    // TODO check login link lifecycle
-    await this._genLoginLink()
+    await new Promise((resolve) => {
+      this.eventTarget.addEventListener('ready', () => {
+        resolve()
+      })
+    })
   }
 
-  async setState (newState) {
-    const currState = this.state
-
-    // BEFORE
-    switch (newState) {
-      case 'notLogged':
-        break
-      case 'userAppDbCreated':
-        break
-      case 'accessMaterialReceived':
-        break
-      case 'registerNeeded':
-        break
-      case 'authorized':
-        break
-      case 'loggingIn':
-        await this._resetLogin(true)
-        break
-      case 'logged':
-        if (currState !== 'notLogged' &&
-            currState !== 'authorized' &&
-            currState !== 'userAppDbCreated') {
-          throw new MasqError(
-            MasqError.INVALID_STATE_TRANSITION,
-            'state transition invalid: ' + currState + ' -> ' + newState)
-        }
-        this._storeSessionInfo()
-
-        break
-      default:
-        throw new MasqError()
-    }
-
-    // TRANSITION
-    debug('[masq.setState] changing state : ' + currState + ' -> ' + newState)
-    this.state = newState
-
-    // AFTER
-    switch (newState) {
-      case 'notLogged':
-        break
-      case 'userAppDbCreated':
-        this._removeDisconnectListener()
-        break
-      case 'accessMaterialReceived':
-        break
-      case 'registerNeeded':
-        break
-      case 'authorized':
-        break
-      case 'loggingIn':
-        break
-      case 'logged':
-        this._removeDisconnectListener()
-
-        debug('[masq.setState] dispatched event: logged_in')
-        this.eventTarget.dispatchEvent(new Event('logged_in'))
-
-        // TODO make sure try/catch is not needed
-        try {
-          this.startReplication()
-        } catch (e) {
-          console.error('startRep error : ' + e)
+  async logIntoMasq (stayConnected) {
+    console.log('[logIntoMasq]')
+    this.service.send({
+      type: 'LOGIN_START',
+      stayConnected
+    })
+    await new Promise((resolve, reject) => {
+      this.service.onTransition(state => {
+        if (state.value === 'logged') {
+          resolve()
+        } else {
```

Comment:
style: `else if` on the same line to keep the block content aligned with the first `if`